default.yml

```diff
@@ -169,17 +160,12 @@ Style/MethodCallWithoutArgsParentheses:
 Style/RaiseArgs:
   Enabled: false
 
-  # Fail is an alias of raise. Avoid aliases, it's more cognitive load for no gain.
+# Fail is an alias of raise. Avoid aliases, it's more cognitive load for no gain.
 # The argument that fail should be used to abort the program is wrong too,
 # there's Kernel#abort for that.
 Style/SignalException:
   EnforcedStyle: only_raise
 
-# { ... } for multi-line blocks is okay, follow Weirichs rule instead:
-# https://web.archive.org/web/20140221124509/http://onestepback.org/index.cgi/Tech/Ruby/BraceVsDoEnd.rdoc
-Style/BlockDelimiters:
-  Enabled: false
```

Comment:
I have not seen Weirich's rule widely used, and gives another level of overhead to considering what style to use. I also personally think multiline braces look weird.

There is an option for `semantic` which I believe is Weirich's rule, so if we want to keep it we could use that instead.

Default (line-based):
```ruby
# bad - single line block
items.each do |item| item / 5 end

# good - single line block
items.each { |item| item / 5 }

# bad - multi-line block
things.map { |thing|
  something = thing.some_method
  process(something)
}

# good - multi-line block
things.map do |thing|
  something = thing.some_method
  process(something)
end
```

Semantic
```ruby
# Prefer `do...end` over `{...}` for procedural blocks.

# return value is used/assigned
# bad
foo = map do |x|
  x
end
puts (map do |x|
  x
end)

# return value is not used out of scope
# good
map do |x|
  x
end

# Prefer `{...}` over `do...end` for functional blocks.

# return value is not used out of scope
# bad
each { |x|
  x
}

# return value is used/assigned
# good
foo = map { |x|
  x
}
map { |x|
  x
}.inspect

# The AllowBracesOnProceduralOneLiners option is ignored unless the
# EnforcedStyle is set to `semantic`. If so:

# If the AllowBracesOnProceduralOneLiners option is unspecified, or
# set to `false` or any other falsey value, then semantic purity is
# maintained, so one-line procedural blocks must use do-end, not
# braces.

# bad
collection.each { |element| puts element }

# good
collection.each do |element| puts element end

# If the AllowBracesOnProceduralOneLiners option is set to `true`, or
# any other truthy value, then one-line procedural blocks may use
# either style. (There is no setting for requiring braces on them.)

# good
collection.each { |element| puts element }

# also good
collection.each do |element| puts element end
```

Reply:
I agree, I tend to use do...ends over the javascript style brackets which are significantly harder to read imho.

Reply:
Same as [1], but I have a strong instinctual preference for `line_count_based`

Reply:
No pref here, I generally only do braces for one liners but in model scope code sometimes braces feels ok.